From 3e6ef9abcad2e645041a28aebb36c99757e0b4f0 Mon Sep 17 00:00:00 2001
From: Fabien Spindler <fabien.spindler@inria.fr>
Date: Wed, 20 May 2026 09:07:08 +0200
Subject: [PATCH 7/10] =?UTF-8?q?Fix=20warning:=20declaration=20of=20?=
 =?UTF-8?q?=E2=80=98fJe=E2=80=99=20shadows=20a=20member=20of=20=E2=80=98vp?=
 =?UTF-8?q?RobotPololuPtu=E2=80=99=20[-Wshadow]?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

visp/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp: In member function ‘virtual void vpRobotPololuPtu::get_fJe(vpMatrix&)’:
visp/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp:73:42: warning: declaration of ‘fJe’ shadows a member of ‘vpRobotPololuPtu’ [-Wshadow]
   73 | void vpRobotPololuPtu::get_fJe(vpMatrix &fJe)
      |                                ~~~~~~~~~~^~~
visp/modules/robot/include/visp3/robot/vpRobot.h:107:12: note: shadowed declaration is here
  107 |   vpMatrix fJe;
      |            ^~~
---
 .../robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp b/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp
index bf1597677..e95666127 100644
--- a/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp
+++ b/modules/robot/src/real-robot/pololu-maestro/vpRobotPololuPtu.cpp
@@ -70,12 +70,12 @@ void vpRobotPololuPtu::get_eJe(vpMatrix &eJe_)
   get_eJe(q, eJe_);
 }

-void vpRobotPololuPtu::get_fJe(vpMatrix &fJe)
+void vpRobotPololuPtu::get_fJe(vpMatrix &fJe_)
 {
   vpColVector q(nDof);
   getPosition(vpRobot::JOINT_STATE, q);

-  get_fJe(q, fJe);
+  get_fJe(q, fJe_);
 }

 void vpRobotPololuPtu::get_eJe(const vpColVector &q, vpMatrix &eJe) const
--
2.43.0
